draft-ietf-quic-transport.md

```diff
@@ -1378,28 +1378,38 @@ A server MUST include the original_connection_id transport parameter
 ({{transport-parameter-definitions}}) if it sent a Retry packet to enable
 validation of the Retry, as described in {{packet-retry}}.
 
-
 ### Values of Transport Parameters for 0-RTT {#zerortt-parameters}
 
-A client that attempts to send 0-RTT data MUST remember the transport parameters
-used by the server.  The transport parameters that the server advertises during
-connection establishment apply to all connections that are resumed using the
-keying material established during that handshake.  Remembered transport
+The transport parameters that the server advertises during connection
+establishment generally apply to all connections that are resumed using
+the keying material established during that handshake.  Remembered transport
 parameters apply to the new connection until the handshake completes and new
 transport parameters from the server can be provided.
 
-A server can remember the transport parameters that it advertised, or store an
-integrity-protected copy of the values in the ticket and recover the information
-when accepting 0-RTT data.  A server uses the transport parameters in
-determining whether to accept 0-RTT data.
-
-A server MAY accept 0-RTT and subsequently provide different values for
-transport parameters for use in the new connection.  If 0-RTT data is accepted
-by the server, the server MUST NOT reduce any limits or alter any values that
-might be violated by the client with its 0-RTT data.  In particular, a server
-that accepts 0-RTT data MUST NOT set values for the following parameters
-({{transport-parameter-definitions}}) that are smaller
-than the remembered value of those parameters.
+The value of the server's previous original_connection_id, preferred_address,
+stateless_reset_token, and ack_delay_exponent MUST NOT be used when
+establishing a new connection; rather, the client should wait to observe the
+server's new values in the handshake.
+
+The client MAY store the server's original max_ack_delay and max_packet_size
```

Comment:
max_ack_delay is not needed for 0-RTT, though it can help in loss recovery, it's a value that can change.

Reply:
The client may find it useful to retain this information to set its 0-RTT loss timer properly, rather than having to wait for the handshake and reset it properly. Am I not thinking clearly on how this is done?

Reply:
No, you are right.  I think that what you have is fine for that one.

Reply:
I realize now what bothered me about max_ack_delay.  This is not something you use in 0-RTT, but it is something you might remember in the same way you might remember the RTT.  It's not guaranteed to be good, but it is certainly better than the defaults for return connections.  You should remove this from this section and maybe add it where we talk about remembering the RTT (if we do that, it's late and I couldn't find it).

Reply:
Why don't you use it in 0RTT? You have to set the loss detection timer, which involves using peer max_ack_delay. If I read the recovery draft correctly?

I made it optional because I suppose someone could update the timer based on the new TPs. The more I think about it, better to just save it.